modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java

```diff
@@ -204,11 +210,85 @@ public Builder length(Integer length) {
             return this;
         }
 
-        /** Builds parameters. */
+        /** Test only builder wrapper. */
+        @TestOnly
         public ColumnParams build() {
+            return build(new ColumnParamsSupplier() {
+                @Override
+                public int getMaxPrecision() {
+                    return 1 << 15;
+                }
+
+                @Override
+                public int getMaxScale() {
+                    return 10;
+                }
+            });
+        }
+
+        /** Builds parameters. */
+        public ColumnParams build(ColumnParamsSupplier helper) {
```

Comment:
I don't think we need ColumnParamsSupplier. The problem with such approach is that it's not up to the caller to decide what max precision should be.

Reply:
@korlov42  may be you show me the right way here ? probably what class or constant i can take ? thnks